synfig-studio/src/gui/widgets/widget_waypoint.cpp

```diff
@@ -294,5 +289,62 @@ Widget_Waypoint::get_waypoint()const
 	return waypoint;
 }
 
+void
+Widget_Waypoint::config_tcb_params(bool show_params)
+{
+	if (show_params) {
+		// show labels
+		tcbFrame->get_label_widget()->show();
+		tensionLabel->show();
+		continuityLabel->show();
+		biasLabel->show();
+		temporalTensionLabel->show();
+		
+		// set the adjustment value
+		adj_tension->set_value(waypoint.get_tension());
+		adj_continuity->set_value(waypoint.get_continuity());
+		adj_bias->set_value(waypoint.get_bias());
+		adj_temporal_tension->set_value(waypoint.get_temporal_tension());
+		
+		// show spin buttons
+		spin_tension->show();
+		spin_continuity->show();
+		spin_bias->show();
+		spin_temporal_tension->show();
+	} else {
+		// hide labels
+		tcbFrame->get_label_widget()->hide();
+		tensionLabel->hide();
+		continuityLabel->hide();
+		biasLabel->hide();
+		temporalTensionLabel->hide();
+		
+		// hide spin buttons
+		spin_tension->hide();
+		spin_continuity->hide();
+		spin_bias->hide();
+		spin_temporal_tension->hide();
+	}
+}
```

Comment:
In order to prevent forgetting some widget to show/hide, prefer to use `Gtk::Widget::set_visible(bool visible = true)`.
This way, only the adjustments will be inside an `if` statement.

Reply:
And maybe you should just show/hide the TCB frame.

Reply:
Another new gtkmm xp gained!
Thanks again :+1: